src/cennznut/v0/contract.rs

```diff
@@ -0,0 +1,293 @@
+// Copyright (C) 2019-2020 Centrality Investments Limited
+//!
+//! # CENNZnut - Contracts
+//!
+//! Delegated smart contract permissioning of CENNZnut for use in CENNZnet
+//!
+
+#![warn(clippy::pedantic)]
+
+#[cfg(not(feature = "std"))]
+extern crate alloc;
+
+#[cfg(feature = "std")]
+extern crate std as alloc;
+
+use bit_reverse::ParallelReverse;
+use codec::{Decode, Encode, Input, Output};
+
+/// A CENNZnet permission domain contract
+#[cfg_attr(test, derive(Clone, Debug, Eq, PartialEq))]
+pub struct Contract {
+    pub address: [u8; 32],
```

Comment:
You could alias `[u8; 32]` as `type ContractAddress = [u8; 32];`

Reply:
Nice!